sentinelhub/download/client.py

```diff
@@ -87,11 +106,11 @@ def download(self, download_requests, max_threads=None, decode_data=True, show_p
                 for future in as_completed(download_list):
                     data_list[future_order[future]] = self._process_download_future(future)
 
-        if is_single_request:
+        if isinstance(download_requests, DownloadRequest):
             return data_list[0]
         return data_list
 
-    def _process_download_future(self, future):
+    def _process_download_future(self, future: Future) -> Any:
```

Comment:
I'm not sure if it's work complicating but this could even be written as:

```Python
def _process_download_future(self, future: Future[T]) -> Optional[T]:
```

Where `Future[T]` works ok for Python >=3.9 and for lower versions probably `typing-extensions` backport can handle it.

Reply:
I dont think it would be of much use, since the download itself returns anything